### PR TITLE
Expose/test Encoder/Decoder span-based APIs

### PR DESCRIPTION
**Pull request by stephentoub**
Depends on https://github.com/dotnet/coreclr/pull/13409 being consumed into corefx.
Fixes https://github.com/dotnet/corefx/issues/23057
cc: @tarekgh

### CODE REVIEW

File: src/System.Text.Encoding/tests/Decoder/Decoder.netcoreapp.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (tarekgh) [inline src/System.Text.Encoding/tests/Decoder/Decoder.netcoreapp.cs]: nit: looks this file testing span APIs only. we may have the file name include Span word (e.g. DecoderSpanTests.netcoreapp.cs)

File: src/System.Text.Encoding/tests/Decoder/Decoder.netcoreapp.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+using Xunit;
+
+namespace System.Text.Encodings.Tests
+{
+    public partial class DecoderTests
+    {
+        [Fact]
+        public static void GetCharCount_Span_MatchesEncodingCharCount()
+        {
+            const string Text = "hello world";
```
Comment (tarekgh) [inline src/System.Text.Encoding/tests/Decoder/Decoder.netcoreapp.cs]: nit: I would prefer to avoid using "Text" as it is part of the encoding namespace

**Comment by tarekgh** [comment]
LGTM. thanks @stephentoub for getting this ready. 

may be later we'll need to add more tests exercise other encodings.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@mellinoe, System.Drawing test failure:
https://ci3.dot.net/job/dotnet_corefx/job/master/job/linux-TGroup_netcoreapp+CGroup_Release+AGroup_x64+TestOuter_false_prtest/2587/

**Comment by stephentoub** [comment]
@dotnet-bot test Windows x64 Debug Build please

**Comment by stephentoub** [comment]
@dotnet-bot test Windows x64 Debug Build please

**Pull request merged**